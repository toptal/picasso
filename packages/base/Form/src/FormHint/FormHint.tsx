import type { ReactNode, HTMLAttributes } from 'react'
import React, { forwardRef } from 'react'
import type { BaseProps } from '@toptal/picasso-shared'
import { Typography } from '@toptal/picasso-typography'
import { twMerge } from '@toptal/picasso-tailwind-merge'

export interface Props extends BaseProps, HTMLAttributes<HTMLDivElement> {
  /** The text of the hint */
  children: ReactNode
}

export const FormHint = forwardRef<HTMLDivElement, Props>(function FormHint(
  props,
  ref
) {
  const { children, className, style, ...rest } = props

  return (
    <div
      {...rest}
      ref={ref}
      className={twMerge('mt-1', className)}
      style={style}
    >
      <Typography size='xxsmall'>{children}</Typography>
    </div>
  )
})

FormHint.defaultProps = {}

FormHint.displayName = 'FormHint'

export default FormHint
