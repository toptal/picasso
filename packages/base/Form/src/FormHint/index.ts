import type { OmitInternalProps } from '@toptal/picasso-shared'

import type { Props } from './FormHint'

export { default as FormHint } from './FormHint'
export type FormHintProps = OmitInternalProps<Props>
