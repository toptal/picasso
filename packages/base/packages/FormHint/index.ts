/* eslint-disable import/no-extraneous-dependencies */
import type { OmitInternalProps } from '@toptal/picasso-shared'

import type { Props } from './FormHint'

export { default } from './FormHint'
export type FormHintProps = OmitInternalProps<Props>
