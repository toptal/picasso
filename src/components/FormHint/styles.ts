import { Theme, createStyles } from '@material-ui/core/styles'

export default ({ palette }: Theme) =>
  createStyles({
    root: {
      marginTop: '.25em'
    },
    hint: {
      color: palette.grey[200],
      fontSize: '.75em'
    }
  })
