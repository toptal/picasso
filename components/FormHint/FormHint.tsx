import React from 'react'
import { withStyles } from '@material-ui/core/styles'

import { Classes } from '../styles/types'
import Typography from '../Typography'
import styles from './styles'

interface Props {
  classes: Classes
  /** The text of the hint */
  children: React.ReactNode
}

export const FormHint: React.FunctionComponent<Props> = props => {
  const { children, classes, ...rest } = props

  return (
    <div className={classes.root} {...rest}>
      <Typography className={classes.hint}>{children}</Typography>
    </div>
  )
}

FormHint.defaultProps = {}

FormHint.displayName = 'FormHint'

export default withStyles(styles)(FormHint)
