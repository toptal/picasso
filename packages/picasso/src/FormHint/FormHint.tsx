import type { ReactNode, HTMLAttributes } from 'react'
import React, { forwardRef } from 'react'
import type { Theme } from '@material-ui/core/styles'
import { makeStyles } from '@material-ui/core/styles'
import cx from 'classnames'
import type { BaseProps } from '@toptal/picasso-shared'

import Typography from '../Typography'
import styles from './styles'

export interface Props extends BaseProps, HTMLAttributes<HTMLDivElement> {
  /** The text of the hint */
  children: ReactNode
}

const useStyles = makeStyles<Theme>(styles, { name: 'FormHint' })

export const FormHint = forwardRef<HTMLDivElement, Props>(function FormHint(
  props,
  ref
) {
  const { children, className, style, ...rest } = props

  const classes = useStyles()

  return (
    <div
      {...rest}
      ref={ref}
      className={cx(classes.root, className)}
      style={style}
    >
      <Typography size='xxsmall'>{children}</Typography>
    </div>
  )
})

FormHint.defaultProps = {}

FormHint.displayName = 'FormHint'

export default FormHint
