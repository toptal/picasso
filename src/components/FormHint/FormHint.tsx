import React, { FunctionComponent, ReactNode } from 'react'
import { withStyles } from '@material-ui/core/styles'
import cx from 'classnames'

import { StandardProps } from '../Picasso'
import Typography from '../Typography'
import styles from './styles'

interface Props extends StandardProps {
  /** The text of the hint */
  children: ReactNode
}

export const FormHint: FunctionComponent<Props> = ({
  children,
  classes,
  className,
  style
}) => (
  <div className={cx(classes.root, className)} style={style}>
    <Typography className={classes.hint}>{children}</Typography>
  </div>
)

FormHint.defaultProps = {}

FormHint.displayName = 'FormHint'

export default withStyles(styles)(FormHint)
