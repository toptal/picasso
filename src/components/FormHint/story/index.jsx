import PicassoBook from '~/.storybook/components/PicassoBook'

import { FormHint } from '../FormHint'

const page = PicassoBook.lookupPage('Form')

page
  .createChapter(
    'Form Hint',
    `Add the <Form.Hint> component if you need to add
    some extra information for your form control`
  )
  .addComponentDocs(FormHint)
  .addExample('FormHint/story/FormHint.example.jsx', 'Form hint')
