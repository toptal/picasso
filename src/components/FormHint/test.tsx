import React from 'react'
import { render, cleanup } from 'react-testing-library'

import FormHint from './index'

const renderFormHint = (text: string) => {
  return render(<FormHint>{text}</FormHint>)
}

afterEach(cleanup)

describe('FormHint', () => {
  test('default render', () => {
    const { container } = renderFormHint(
      'Picasso was born on October 25, 1881 in the city of Malaga, Spain'
    )

    expect(container).toMatchSnapshot()
  })
})
