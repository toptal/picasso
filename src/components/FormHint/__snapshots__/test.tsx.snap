// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHint default render 1`] = `
<div>
  <div
    class="FormHint-root"
  >
    <p
      class="MuiTypography-root Typography-body Typography-regular MuiTypography-body1 FormHint-hint"
    >
      Picasso was born on October 25, 1881 in the city of Malaga, Spain
    </p>
  </div>
</div>
`;
